src/main/java/org/jabref/logic/shared/DBMSProcessor.java

```diff
@@ -431,6 +431,47 @@ public void removeEntry(BibEntry bibEntry) {
         return getSharedEntryList(0);
     }
 
+    public List<BibEntry> getSharedEntriesByIdList(List<Integer> idList) {
+        List<BibEntry> sharedEntries = new ArrayList<>();
+
+        StringBuilder query = new StringBuilder();
+        query.append("SELECT * from ")
+                .append(escape("ENTRY"))
+                .append(" inner join ")
+                .append(escape("FIELD"))
+                .append(" F on ")
+                .append(escape("ENTRY") + ".")
+                .append(escape("SHARED_ID") + " = F." + escape("ENTRY_SHARED_ID"))
+                .append(" where ")
+                .append(escape("SHARED_ID")).append(" in (");
+
+        for (int i = 0; i < idList.size() - 1; i++) {
+            query.append(idList.get(i)).append(", ");
+        }
+        query.append(idList.get(idList.size() - 1)).append(") order by ").append(escape("SHARED_ID")).append(";");
+
```

Comment:
Result query will be 
`SELECT *
from ENTRY
right join FIELD F on ENTRY.SHARED_ID = F.ENTRY_SHARED_ID
where SHARED_ID in (7, 4, 9) order by SHARED_ID;`

Reply:
Ah I see! Just another optimization hint: Never use `*` in a query. You should add all columns instead.

Reply:
Ok. I will fix it.

Reply:
FIxed